ts/IterableInbox.tsx

```diff
@@ -1,22 +1,39 @@
 'use strict';
-import React, { Component, useState } from 'react';
-import { Text, View, StyleSheet } from 'react-native';
-import IterableInboxMessageList from './IterableInboxMessageList';
-import IterableInboxEmptyState from './IterableInboxEmptyState';
-import sampleMessages from './sampleMessageData.js';
+import React, { Component, useState } from 'react'
+import { Text, View, StyleSheet } from 'react-native'
+import IterableInboxMessageList from './IterableInboxMessageList'
+import IterableInboxEmptyState from './IterableInboxEmptyState'
+import IterableInboxMessageDisplay from './IterableInboxMessageDisplay'
+import sampleMessages from './sampleMessageData.js'
 
 const IterableInbox = () => {
-   const inboxTitle = "Inbox";
-   const [selectedMessageIdx, setSelectedMessageIdx] = useState(0);
-   const [messages, setMessages] = useState(sampleMessages);
+   const inboxTitle = "Inbox"
+   const [isDisplayMessage, setIsDisplayMessage] = useState(false)
+   const [selectedMessageIdx, setSelectedMessageIdx] = useState(0)
+   const [messages, setMessages] = useState(sampleMessages)
+
+   let selectedMessage = messages[selectedMessageIdx]
 
    function handleMessageSelect(index: number, messages: Array<any>) {
-      if(!messages[index].read) {
+      if (!messages[index].read) {
          messages[index].read = true
       }
-      setMessages([...messages]);
-      setSelectedMessageIdx(index);
-   }   
+      setMessages([...messages])
```

Comment:
this all feels a little bit off to me. it may actually provided the intended result but it does so inspite of how react is built. 

when you are doing the following: 
```
if (!messages[index].read) {
    messages[index].read = true
}
```
you are actually making a mutable change to the messages object. which is why you then have to de-structure the `messages` array in the `setMessages` call for the state to get updated.  

as i said above, this all technically accomplishes what you're trying to do, but is a bit roundabout. 

instead, how do you feel about something like the following: 
```
function handleMessageSelect(index: number, messages: Array<any>) {
   const newMessages = messages.map((message, messageIndex) => {
      return (messageIndex === index) ?
         { ...message, read: true } : message
    }
   setMessages(newMessages);
   setIsDisplayMessage(true);
   setSelectedMessageIdx(index);
}
```

Reply:
I agree. This is a better approach. I should not be changing the state directly.